controllers/fluent_controller_finalizer.go

```diff
@@ -108,12 +108,15 @@ func (r *FluentdReconciler) mutate(obj client.Object, fd *fluentdv1alpha1.Fluent
 			}
 			return nil
 		}
-	case *appsv1.Deployment:
-		expected := operator.MakeDeployment(*fd)
+	case *appsv1.StatefulSet:
+		expected := operator.MakeStatefulset(*fd)
 
 		return func() error {
 			o.Labels = expected.Labels
-			o.Spec = expected.Spec
+			o.Spec.Replicas = expected.Spec.Replicas
+			o.Spec.Template = expected.Spec.Template
+			o.Spec.UpdateStrategy = expected.Spec.UpdateStrategy
+			o.Spec.VolumeClaimTemplates = expected.Spec.VolumeClaimTemplates
```

Comment:
sts use `VolumeClaimTemplates`, do we still need `case PersistentVolumeClaim` above?

Reply:
I think no need. I will fix it and have a test.